### DISCUSSION TITLE
Questions on using Add-PnPDataRowsToProvisioningTemplate

### DISCUSSION THREAD

**Discussion by renewtx** [pnp/powershell#3537] [Q&A]
Hello,

I am working on a provisioning script for SharePoint 2016 on-prem (using latest version of SharePointPnPPowerShell2016).  So far everything is working as expected but I am encountering a couple issues with Add-PnPDataRowsToProvisioningTemplate ...

===================================================================================
First, I am using Get-PnPProvisioningTemplate cmdlet to generate a set of template files from my source/template site.  Once I have these component template files, I use Apply-PnPProvisioningTemplate to apply all of these template files against my target site to properly provision and configure it.  All of this works pretty well so far except for a couple of issues I am trying to fix.

First, using the below I generate a template file for my lists that I want from the template site:
$handlers = "Lists"
Get-PnPProvisioningTemplate -Out ($workingFolder + "Template-4-SPLIST.xml") -Handlers $handlers -IncludeHiddenLists

This generates the Template-4-SPLIST.xml as expected.  I then use this cmdlet to add list items from a couple of lists into my template:
Add-PnPDataRowsToProvisioningTemplate -Path ($workingFolder + "Template-4-SPLIST.xml") -List "Promoted Links". 

For the Promoted Links (std SP list) list this works fine and when I check the XML file, I do see a <pnp:DataRow> element get added for this list and it is correct.  However, when I run this template against my target site, the Promoted Links list gets added, but the data row is not added ... Is there some trick to getting list items specified via the DataRow element added to the list in the target site?

===================================================================================

Second, I also have a Site Contacts list (again the std SP list) and I want to also include it (and one entry in this list) to be added to the target site, but when I run the same cmdlet as above

Add-PnPDataRowsToProvisioningTemplate -Path ($workingFolder + "Template-4-SPLIST.xml") -List "Site Contacts"

I get this error:

Add-PnPDataRowsToProvisioningTemplate : The property or field 'Id' has not been initialized. It has not been requested or the request has not been executed. It may need to be explicitly 
requested.
At line:1 char:1
+ Add-PnPDataRowsToProvisioningTemplate -Path ($workingFolder + " Template-4 ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Add-PnPDataRowsToProvisioningTemplate], PropertyOrFieldNotInitializedException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Provisioning.Site.AddDataRowsToProvisioningTemplate

===================================================================================

Any thoughts or ideas appreciated.

Thanks

**Comment by AndersRask** [comment]
This repo covers PnP.PowerShell, not the legacy repos (that is deprecated).
You can post SharePointPnPPowerShell2016 related questions [here](https://github.com/pnp/PnP-PowerShell/discussions)

**Reply by renewtx** [reply]
It looks like that repository was made read-only/archived in Jan 2021 ... any other places where I could post this question??
